libres/lib/enkf/enkf_main.cpp

```diff
@@ -699,56 +696,59 @@ int enkf_main_load_from_run_context(enkf_main_type *enkf_main,
     auto const ens_size = enkf_main_get_ensemble_size(enkf_main);
     auto const *iactive = ert_run_context_get_iactive(run_context);
 
-    int result[ens_size];
-    arg_pack_type **arg_list = (arg_pack_type **)util_calloc(
-        ens_size, sizeof *arg_list); // CXX_CAST_ERROR
-    thread_pool_type *tp =
-        thread_pool_alloc(std::thread::hardware_concurrency(), true);
+    // Loading state from a fwd-model is mainly io-bound so we can
+    // allow a lot more than #cores threads to execute in parallel.
+    // The number 100 is quite arbitrarily chosen though and should
+    // probably come from some resource like a site-config or similar.
+    // NOTE that this mechanism only limits the number of *concurrently
+    // executing* threads. The number of futures instantiated and stored
+    // will be equal to the number of active realizations.
+    Semafoor max_jobs(100);
```

Comment:
Does the name `max_jobs` make sense?

Reply:
Perhaps `max_running` or `simultaneous_executing_threads` ?

Reply:
Suggested `max_concurrent_threads` in new version

Reply:
My issue with the name is that it suggests that it's upper limit of threads, rather than a semaphore object. (Ie. it's the `max` part that I find potentially confusing for future generations).

Perhaps `concurrency_lock` or `concurrency_limiter`? A good name will potentially not require the comment prior.

Anyway, this is a fairly minor point.